examples/single_column_carbon_alkalinity_nutrients.jl

```diff
@@ -68,6 +115,92 @@ simulation = Simulation(model, Δt=10minutes, stop_time=12days)
 progress(sim) = @printf("Iteration: %d, time: %s\n", iteration(sim), prettytime(sim))
 simulation.callbacks[:progress] = Callback(progress, IterationInterval(10))
 
+# I really want the option to take these from the model
+Base.@kwdef struct Pᶠˡᵘˣᶜᵒ²{FT}
+    open_water_fraction :: FT = 1.
+    surface_wind_speed  :: FT = 5. # ms⁻¹
+    atmospheric_pCO₂    :: FT = 280e-6 # atm
+    exchange_coefficient:: FT = 0.337 # cm/hr
+    silicate_conc       :: FT = 15e-6 # umol/kg
+    initial_pH_guess    :: FT = 8.0
+end
+
+# Build operation for CO₂ flux calculation, used in callback below to calculate every time step
+"""
+    compute_co₂_flux!(simulation)
+    
+Returns the tendency of DIC in the top layer due to air-sea CO₂ flux
+using the piston velocity formulation of Wanninkhof (1992) and the
+solubility/activity of CO₂ in seawater.
+"""
+@inline function compute_co₂_flux!(simulation)
+# applied pressure in atm (i.e. Pˢᵘʳᶠ-Pᵃᵗᵐ)
+    Δpᵦₐᵣ           = 0.0
+
+# conversion factor from cm/hr to m/s
+    cmhr⁻¹_per_ms⁻¹ = 1 / 3.6e5
+
+# get coefficients from Pᶠˡᵘˣᶜᵒ² struct
+# I really want the option to take these from the model
```

Comment:
an ocean model doesn't have atmospheric pCO2 though... we need a coupled model for that

Reply:
It does if it wants to calculate air-sea CO<sub>2</sub> fluxes. Until we reach the coupled-model stage, we'll have to prescribe it or set up a simple conservative "box".

Reply:
Sorry, I mean by definition, an Oceananigans ocean model does not have atmospheric pCO2. This is our design.

Reply:
I suppose the actual definition is the pCO<sub>2</sub> of seawater in equilibrium with the atmosphere, which is why pCO₂ᵃᵗᵐ is multiplied by a solubility coefficient. But we still require a value for atmospheric pCO<sub>2</sub> to calculate that.

Also see [`gasexchange`](https://github.com/OceanBioME/OceanBioME.jl/blob/12cfa7fd9edb95a5058cab9c32983285d2ddb952/src/Boundaries/gasexchange.jl#L196C1-L206C107) in `OceanBiome.jl`, which has `air_concentration::AC = (CO₂ = 413.4, O₂ = 9352.7)[gas]`.

It's a totally routine thing to do in ocean-only carbon cycle models.

Reply:
I'm not sure I understand what you're saying. Do you mean we should add a property `atmospheric_pCO2` to `HydrostaticFreeSurfaceModel`?

In `ClimaOcean`, we're designing an `OceanSeaIceModel`. This object contains references to:

1. An ocean `Simulation`
2. A sea ice model (optionally, this is `nothing`, or it might be either diagnostic or prognostic)
3. A `PrescribedAtmosphere`.

When we use `OceanSeaIceModel`, we compute fluxes of momentum, heat, moisture, and hopefully eventually trace gases using bulk formula and the state of `PrescribedAtmosphere` at the relevant space and time.

I think that `PrescribedAtmosphere` should eventually also support specified CO2 or concentrations? And every other trace gas?

For idealized cases where we don't require the full atmospheric state for some reason, at the moment users have to write up their own setups.

`OceanBioME` goes it's own way to some degree. I think we have benefited in the past from taking a systematic approach. While true that our user interface and philosophy is innovative, I think it will both prove to be much more productive for most people, and also more extensible (easier to develop).

That said you are free to propose an alternative approach for sure. But I think these suggestions should come from a detailed understanding of the existing data structures and approach.